locksmith/__tests__/operations/userOperations.test.ts

```diff
@@ -0,0 +1,86 @@
+import UserOperations from '../../src/operations/userOperations'
+
+const models = require('../../src/models')
+let User: any = models.User
+let UserReference: any = models.UserReference
+
+describe('User creation', () => {
+  let userCreationDetails = {
+    emailAddress: 'user@example.com',
+    publicKey: '0x21cC9C438D9751A3225496F6FD1F1215C7bd5D83',
+    passwordEncryptedPrivateKey: '{"data" : "encryptedPassword"}',
+    recoveryPhrase: 'recoveryPhrase',
+  }
+
+  describe("when a user for the public key doesn't exist", () => {
+    it('should create a user and associated data', async () => {
+      UserReference.create = jest.fn(() => {})
+
+      expect.assertions(1)
+      await UserOperations.createUser(userCreationDetails)
+      expect(UserReference.create).toHaveBeenCalledWith(
+        {
+          User: {
+            passwordEncryptedPrivateKey: '{"data" : "encryptedPassword"}',
+            publicKey: '0x21cC9C438D9751A3225496F6FD1F1215C7bd5D83',
+            recoveryPhrase: 'recoveryPhrase',
+          },
+          emailAddress: 'user@example.com',
+        },
+        { include: User }
+      )
+    })
+
+    it('should normalize the public key/address', async () => {
+      UserReference.create = jest.fn(() => {})
+      expect.assertions(1)
+      await UserOperations.createUser(userCreationDetails)
+      expect(UserReference.create).toHaveBeenCalledWith(
+        {
+          User: {
+            passwordEncryptedPrivateKey: '{"data" : "encryptedPassword"}',
+            publicKey: '0x21cC9C438D9751A3225496F6FD1F1215C7bd5D83',
```

Comment:
Could you actually denormalize (lowercase) the `publicKey` in the `userCreationDetails` so that we know we're actually normalzing it?